packages/flutter_tools/lib/src/migrations/swift_package_manager_integration_migration.dart

```diff
@@ -557,69 +565,90 @@ $newContent
-    if (runnerNativeTarget == null) {
-      throw Exception(
-        'Unable to find parsed PBXNativeTarget for ${_xcodeProject.hostAppProjectName} target.',
-      );
-    }
+  void _migrateNativeApplicationTarget(
+    List<String> lines,
+    ParsedNativeTarget nativeTarget, {
+    required int startSectionIndex,
+    required int endSectionIndex,
+  }) {
+    final String targetName = nativeTarget.name ?? '';
+    final String pbxTargetName =
+        targetName.isNotEmpty ? 'PBXNativeTarget "$targetName"' : 'PBXNativeTarget';
     final String subsectionLineStart =
-        runnerNativeTarget.name != null
-            ? '$_runnerNativeTargetIdentifier /* ${runnerNativeTarget.name} */ = {'
-            : _runnerNativeTargetIdentifier;
-    final int runnerNativeTargetStartIndex = lines.indexWhere(
+        targetName.isNotEmpty
+            ? '${nativeTarget.identifier} /* $targetName */ = {'
+            : nativeTarget.identifier;
+    final int nativeTargetStartIndex = lines.indexWhere(
       (String line) => line.trim().startsWith(subsectionLineStart),
       startSectionIndex,
     );
-    if (runnerNativeTargetStartIndex == -1 || runnerNativeTargetStartIndex > endSectionIndex) {
+
+    if (nativeTargetStartIndex == -1 || nativeTargetStartIndex > endSectionIndex) {
       throw Exception(
-        'Unable to find PBXNativeTarget for ${_xcodeProject.hostAppProjectName} target.',
+        'Unable to find $pbxTargetName for ${_xcodeProject.hostAppProjectName} project.',
       );
     }
 
-    if (runnerNativeTarget.packageProductDependencies == null) {
+    if (nativeTarget.packageProductDependencies == null) {
       // If packageProductDependencies is null, the packageProductDependencies field is missing and must be added.
       const List<String> newContent = <String>[
         '			packageProductDependencies = (',
         '				$_flutterPluginsSwiftPackageProductDependencyIdentifier /* FlutterGeneratedPluginSwiftPackage */,',
         '			);',
       ];
-      lines.insertAll(runnerNativeTargetStartIndex + 1, newContent);
-    } else {
-      // Find the packageProductDependencies field within the Native Target for the Runner target.
-      final int packageProductDependenciesIndex = lines.indexWhere(
-        (String line) => line.trim().contains('packageProductDependencies'),
-        runnerNativeTargetStartIndex,
+      lines.insertAll(nativeTargetStartIndex + 1, newContent);
+      return;
+    }
+
+    // Find the packageProductDependencies field within the Native Target.
+    final int packageProductDependenciesIndex = lines.indexWhere(
+      (String line) => line.trim().contains('packageProductDependencies'),
+      nativeTargetStartIndex,
+    );
+    if (packageProductDependenciesIndex == -1 ||
+        packageProductDependenciesIndex > endSectionIndex) {
+      throw Exception(
+        'Unable to find packageProductDependencies for $pbxTargetName in ${_xcodeProject.hostAppProjectName} project.',
```

Comment:
Added the name of the affected target + clarifying that hostAppProjectName is the project